k8s/upgrades/0.9.0-1.0.0/jiva/jiva_volume_upgrade.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+
+target_upgrade_version="1.0.0-RC1"
+current_version="0.9.0"
+
+function usage() {
+    echo
+    echo "Usage:"
+    echo
+    echo "$0 <pv-name>"
+    echo
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+    usage
+fi
+
+pv=$1
+
+# Check if pv exists
+kubectl get pv $pv &>/dev/null;check_pv=$?
+if [ $check_pv -ne 0 ]; then
+    echo "$pv not found";exit 1;
+fi
+
+# Check if CASType is jiva
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}"`
+if [ $cas_type != "jiva" ]; then
+    echo "Jiva volume not found";exit 1;
+else [ $cas_type == "jiva" ];
+    echo "$pv is a jiva volume"
+fi
+
+ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+sc_name=`kubectl get pv $pv -o jsonpath="{.spec.storageClassName}"`
+sc_res_ver=`kubectl get sc $sc_name -n $ns -o jsonpath="{.metadata.resourceVersion}"`
+
+#################################################################
+# STEP: Generate deploy, replicaset and container names from PV #
+#                                                               #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc"   #
+#                                                               #
+# ctrl-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-ctrl       #
+#################################################################
+
+c_dep=$(kubectl get deploy -n $ns -l openebs.io/persistent-volume=$pv,openebs.io/controller=jiva-controller -o jsonpath="{.items[*].metadata.name}")
+r_dep=$(kubectl get deploy -n $ns -l openebs.io/persistent-volume=$pv,openebs.io/replica=jiva-replica -o jsonpath="{.items[*].metadata.name}")
+c_svc=$(kubectl get svc -n $ns -l openebs.io/persistent-volume=$pv -o jsonpath="{.items[*].metadata.name}")
+c_name=$(kubectl get deploy -n $ns $c_dep -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" | grep "ctrl-con")
+r_name=$(kubectl get deploy -n $ns $r_dep -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" | grep "rep-con")
+
+# Fetch the older target and replica - ReplicaSet objects which need to be
+# deleted before upgrading. If not deleted, the new pods will be stuck in
+# creating state - due to affinity rules.
+
+c_rs=$(kubectl get rs -o name --namespace $ns -l openebs.io/persistent-volume=$pv,openebs.io/controller=jiva-controller | cut -d '/' -f 2)
```

Comment:
can we rename this to `c_rs_old` & next one as `r_rs_old`

Reply:
done